tests/unit/sys/test_options.cxx

```diff
@@ -35,6 +37,24 @@ TEST_F(OptionsTest, SetGetInt) {
   EXPECT_EQ(value, 42);
 }
 
+TEST_F(OptionsTest, SetGetIntFromReal) {
+  Options options;
+  options.set("int_key", 42.00001, "code");
+
+  ASSERT_TRUE(options.isSet("int_key"));
+
+  int value;
+  options.get("int_key", value, 99, false);
+
+  EXPECT_EQ(value, 42);
+
+  options.set("int_key2", 12.5, "code");
+  EXPECT_THROW(options.get("int_key2", value, 99, false), BoutException);
+  // Note we expect to get the rounded value despite the throw as we
+  // pass by reference and modify the passed variable in options.get.
+  EXPECT_EQ(value, 13);
```

Comment:
Is this what we actually *want* to happen though? This feels more like a bug to me

Reply:
It's only because we catch the throw here though and carry on that we see this at all. It's slightly unexpected but I don't think there is an expected value that makes sense either.

Reply:
True. In that case, we should maybe not test the value after, and say that it's "undefined" instead.

Reply:
We could -- I mostly included this as some form of documentation.
If we want the value to be undefined then I wonder if we can make `get` force the value to be `NaN` to clearly indicate our intent?

It's vaguely possible that someone's code is relying on this (without realising it) so by including this test we'd at least catch when this assumption changes.